lib/web/apiserver.go

```diff
@@ -2701,3 +2707,21 @@ func ssoSetWebSessionAndRedirectURL(w http.ResponseWriter, r *http.Request, resp
 
 	return nil
 }
+
+// isRoleMatchError checks if an error is the result of a failed role match during
+// SSO authorization.
+func isRoleMatchError(err error) bool {
+	// oidc or saml
+	if trace.IsAccessDenied(err) {
+		// from lib/auth/saml.go:calculateSAMLUser
```

Comment:
I thought we didn't control the error strings, but if they are part of our codebase then let's ditch the comments and make sure they stay in sync (through a constant string, constant error, error.Is check, etc).